### PR TITLE
Add types for ajv-merge-patch

### PR DESCRIPTION
**Pull request by ghost**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

---

Moved from epoberezkin/ajv-merge-patch#29

### CODE REVIEW

File: types/ajv-merge-patch/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Type definitions for ajv-merge-patch 4.1
+// Project: https://github.com/epoberezkin/ajv-merge-patch#readme
+// Definitions by: Zhu Zijia <https://github.com/littlepiggy03>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+import { Ajv } from "ajv";
+
+declare namespace ajvMergePatch {}
```
Comment (plantain-00) [inline types/ajv-merge-patch/index.d.ts]: I think this line can be removed

File: types/ajv-merge-patch/ajv-merge-patch-tests.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import * as Ajv from "ajv";
+import * as ajvMergePatch from "ajv-merge-patch";
```
Comment (plantain-00) [inline types/ajv-merge-patch/ajv-merge-patch-tests.ts]: Should use `import ajvMergePatch = require("ajv-merge-patch");` here

File: types/ajv-merge-patch/tslint.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
```
Comment (plantain-00) [inline types/ajv-merge-patch/tslint.json]: This file should only `"extends": "dtslint/dt.json"`

**Review by plantain-00 — CHANGES_REQUESTED** [review]

File: types/ajv-merge-patch/ajv-merge-patch-tests.ts (hunk 2 above)
Comment (littlepiggy03) [reply types/ajv-merge-patch/ajv-merge-patch-tests.ts]: May I ask why should remove `declare namespace ajvMergePatch {}` and use `import ajvMergePatch = require("ajv-merge-patch");`?

Comment (plantain-00) [reply types/ajv-merge-patch/ajv-merge-patch-tests.ts]: https://github.com/DefinitelyTyped/DefinitelyTyped#should-i-add-an-empty-namespace-to-a-package-that-doesnt-export-a-module-to-use-es6-style-imports

Comment (littlepiggy03) [reply types/ajv-merge-patch/ajv-merge-patch-tests.ts]: Thanks! :stuck_out_tongue_closed_eyes:

**Review by plantain-00 — APPROVED** [review]

**Pull request merged**